scripts/deleteSnapsets.js

```diff
@@ -0,0 +1,243 @@
+/*!
+Copyright 2018 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+// This script modifies the preferences data base:
+// 1. Finds all the Prefs Safes of type "snapset" (prefsSafesType = "snapset"),
+// 2. Finds all the GPII Keys associated with each snapset Prefs Safe
+// 3. Deletes the found Prefs Safes and associated GPII Keys
+//
+// A sample command that runs this script:
+// node deleteSnapsets.js $COUCHDBURL
+
+"use strict";
+
+var http = require("http"),
+    url = require("url"),
+    fluid = require("infusion");
+
+var gpii = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.dataLoader");
+fluid.setLogging(fluid.logLevel.INFO)
+
+var dbLoader = gpii.dataLoader;
+dbLoader.couchDbUrl = process.argv[2];
+if (!fluid.isValue(dbLoader.couchDbUrl)) {
+    fluid.log("COUCHDB_URL environment variable must be defined");
+    fluid.log("Usage:  node deleteSnapsets.js $COUCHDB_URL");
+    process.exit(1);
+}
+dbLoader.prefsSafesViewUrl = dbLoader.couchDbUrl + "/_design/views/_view/findSnapsetPrefsSafes";
+dbLoader.gpiiKeysViewUrl = dbLoader.couchDbUrl + "/_design/views/_view/findAllGpiiKeys";
+dbLoader.parsedCouchDbUrl = url.parse(dbLoader.couchDbUrl);
+dbLoader.snapsetPrefsSafes = [];
+dbLoader.gpiiKeys = [];
+
+fluid.log("COUCHDB_URL: '" +
+    dbLoader.parsedCouchDbUrl.protocol + "//" +
+    dbLoader.parsedCouchDbUrl.hostname + ":" +
+    dbLoader.parsedCouchDbUrl.port +
+    dbLoader.parsedCouchDbUrl.pathname +
+"'");
+
+/**
+ * Find the Prefs Safes of type "snapset", mark them to be deleted and add
+ * them to an array of records to remove.
+ * @param {String} responseString - The response from the database query for
+ *                                  retrieving the snapset PrefsSafes records
+ * @return {Array} - Array of snapset Prefs Safes each with their "deleted"
+ *                   field set to true.
+ */
+dbLoader.processSnapsets = function (responseString) {
+    fluid.log("Processing the snapset Prefs Safes records...");
+    dbLoader.snapSets = JSON.parse(responseString);
+    fluid.each(dbLoader.snapSets.rows, function (aSnapset) {
+        aSnapset.value._deleted = true;
+        dbLoader.snapsetPrefsSafes.push(aSnapset.value);
+    });
+    fluid.log("\tSnapset Prefs Safes marked for deletion.");
+    return dbLoader.snapsetPrefsSafes;
```

Comment:
This line seems unnecessary because `dbLoader.snapsetPrefsSafes` is a global variable.

Reply:
... and nothing actually uses the return value.  Removed.